packages/entities/entities-shared/src/composables/useDebouncedFilter.ts

```diff
@@ -93,20 +98,58 @@ export default function useDebouncedFilter(
         error.value = ''
         validationError.value = ''
 
-        // If user has typed info in the query field
-        let currUrl = url + '' // clone
-        if (query) {
-          currUrl += `/${query}`
-        }
+        if (config.app === 'konnect') { // KoKo only supports exact match
+          // If user has typed info in the query field
+          let currUrl = url + '' // clone
+          if (query) {
+            currUrl += `/${query}`
+          }
 
-        const { data }: Record<string, any> = await axiosInstance.get(`${currUrl}?size=${size}`)
+          const { data }: Record<string, any> = await axiosInstance.get(`${currUrl}?size=${size}`)
 
-        if (keys.fetchedItemsKey in data) {
-          results.value = data[keys.fetchedItemsKey]
-        } else if (data?.id) { // exact match
-          results.value = [data]
-        } else {
-          results.value = []
+          if (keys.fetchedItemsKey in data) {
+            results.value = data[keys.fetchedItemsKey]
+          } else if (data?.id) { // exact match
+            results.value = [data]
+          } else {
+            results.value = []
+          }
+        } else { // Admin API supports filtering on specific fields
+          const promises = []
+
+          if (isValidUuid(query || '') && keys.searchKeys.includes('id')) {
+            // If query is a valid UUID, do the exact search
+            promises.push((async () => {
+              const res: Record<string, any> = await axiosInstance.get(`${url}/${query}`)
+              return [res[keys.fetchedItemsKey]]
```

Comment:
Not sure but maybe it should be `return [res.data]` here. I think it happens to work because the `fetchedItemsKey` defaults to `"data"`.

Reply:
You are right. I changed this to
``` js
const { data } = await axiosInstance.get(`${url}/${query}`)
return [data[keys.fetchedItemsKey] ?? data]
```